apis/v1alpha2/validation/tlsroute_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+func TestValidateTLSRoute(t *testing.T) {
+	t.Parallel()
+
+	portNumber := gatewayv1a2.PortNumber(9080)
+
+	tests := []struct {
+		name  string
+		rules []gatewayv1a2.TLSRouteRule
+		errs  field.ErrorList
+	}{
+		{
+			name: "valid TLSRoute with 1 backendRef",
+			rules: []gatewayv1a2.TLSRouteRule{
+				{
+					BackendRefs: []gatewayv1a2.BackendRef{
+						{
+							BackendObjectReference: gatewayv1a2.BackendObjectReference{
+								Port: &portNumber,
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
:-) lets use create a `makeRoute()` helper and call it. This is really hard to read.

Reply:
https://github.com/kubernetes-sigs/gateway-api/pull/1601 added this in 168963d446a5bacd5e6400ebe363670af1153542